XlsxFile.js

```diff
@@ -0,0 +1,244 @@
+ * the given project.
+ *
+ * @param {Project} project the project object
+ * @param {String} pathName path to the file relative to the root
+ * of the project file
+ * @param {FileType} type the file type of this instance
+ */
+var XlsxFile = function(props) {
+    var langDefaultLocale;
+
+    this.baseLocale = false;
+    this.project = props.project;
+    this.pathName = props.pathName;
+    this.locale = new LocaleMatcher({locale:props.locale}).getLikelyLocaleMinimal();
+    this.API = props.project.getAPI();
+
+    langDefaultLocale = new LocaleMatcher({locale: this.locale.language});
+    this.baseLocale = langDefaultLocale.getLikelyLocaleMinimal().getSpec() === this.locale.getSpec();
+    this.type = props.type;
+
+    this.datatype = "x-xlsx";
+    this.set = this.API.newTranslationSet(this.project ? this.project.sourceLocale : "zxx-XX");
+};
+
+/**
+ * Unescape the string to make the same string that would be
+ * in memory in the target programming language.
+ *
+ * @static
+ * @param {String} string the string to unescape
+ * @returns {String} the unescaped string
+ */
+XlsxFile.unescapeString = function(string) {
+    if (!string) return;
+    var unescaped = string;
+
+    unescaped = unescaped.
+        replace(/\\\\n/g, "").                // line continuation
+        replace(/\\\n/g, "").                // line continuation
+        replace(/^\\\\/, "\\").             // unescape backslashes
+        replace(/([^\\])\\\\/g, "$1\\").
+        replace(/^\\'/, "'").               // unescape quotes
+        replace(/([^\\])\\'/g, "$1'").
+        replace(/^\\"/, '"').
+        replace(/([^\\])\\"/g, '$1"');
+
+    return unescaped;
+};
+
+/**
+ * Clean the string to make a resource name string. This means
+ * removing leading and trailing white space, compressing
+ * whitespaces, and unescaping characters. This changes
+ * the string from what it looks like in the source
+ * code but increases matching.
+ *
+ * @static
+ * @param {String} string the string to clean
+ * @returns {String} the cleaned string
+ */
+XlsxFile.cleanString = function(string) {
+    if (!string) return;
+    var unescaped = XlsxFile.unescapeString(string);
+
+    unescaped = unescaped.
+        replace(/\\[btnfr]/g, " ").
+        replace(/[ \n\t\r\f]+/g, " ").
+        trim();
+
+    return unescaped;
+};
+
+/**
+* Add a resource to this file. The locale of the resource
+* should correspond to the locale of the file, and the
+* context of the resource should match the context of
+* the file.
+*
+* @param {Resource} res a resource to add to this file
+*/
+XlsxFile.prototype.addResource = function(res) {
+    logger.trace("XlsxFile.addResource: " + JSON.stringify(res) + " to " + this.project.getProjectId() + ", " + this.locale + ", " + JSON.stringify(this.context));
+    var resLocale = res.getTargetLocale() || res.getSourceLocale();
+
+    if (res && res.getProject() === this.project.getProjectId() && resLocale === this.locale.getSpec()) {
```

Comment:
This plugin only supports 1 locale at a time? This line enforces that...

Reply:
yes. Including this, all other webOS plugins also have the same implementation so far